src/web100srv_unit_tests.c

```diff
@@ -8,21 +9,41 @@
 
 #include "unit_testing.h"
 
-/** Starts the web100srv process. */
-pid_t start_server(int port) {
+/** Starts the web100srv process. Has a variable number of arguments, the last
+ * of which should be null. */
+pid_t start_server(int port, ...) {
+  va_list arglist;
+  int rv;
+  char *args_for_exec[256] = {"./web100srv", "--port", NULL};
+  int i = 2;  // the first index of args_for_exec that is not NULL
+  char *arg;
   pid_t server_pid;
-  int server_return_value;
+  siginfo_t server_status;
   char port_string[] = "32767";  // max port
   fprintf(stderr, "Starting the server\n");
   sprintf(port_string, "%d", port);
   if ((server_pid = fork()) == 0) {
-    execl("./web100srv", "./web100srv", "--port", port_string, NULL);
+    args_for_exec[i++] = port_string;
```

Comment:
Can we the `sprintf` of `port_string` to this scope since it's only used in here?